### PR TITLE
[MRG] Relax the PR checklist

### LINKED ISSUE #1822: Revisit PR checklist
Our PR checklist is pretty heavyweight right now, and is routinely ignored by PR submitters. I too often find myself deleting or <strike>crossing out</strike> many of the checklist items.

If the checklist is going to be ignored, then it's worse than useless. Do we need to trim it down to make it less tedious? Should we be stricter about enforcing checklist completion before code review? I'm +1 on both.

### PR DESCRIPTION
**Pull request by standage**
Following up on #1822, this PR updates the PR checklist and removes several items that are being ignored or have been satisfied in other ways. I've attempted to make the new checklist descriptive as to our current process, rather than prescriptive about what that process should be. I think it's a fairly reasonable compromise between what we should expect of new contributors and what we should remind frequent contributors to do without too much chafing.

Closes #1822.

Ready for discussion (and review and merge)!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,16 +1,27 @@
-- [ ] Is it mergeable?
-- [ ] `make test` Did it pass the tests?
-- [ ] `make clean diff-cover` If it introduces new functionality in
-  `scripts/` is it tested?
```
Comment (standage) [inline .github/PULL_REQUEST_TEMPLATE.md]: No need to burden the users with these first 3 points. The GitHub interface won't allow unmergeable merges, Travis CI runs the test build, and CodeCov gives a coverage report.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
-- [ ] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
-  formatted?
```
Comment (standage) [inline .github/PULL_REQUEST_TEMPLATE.md]: Any `pep8` violations will cause a CI build failure. Anything beyond this hasn't been well-enforced up to this point.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
-- [ ] Did it change the command-line interface? Only backwards-compatible
-  additions are allowed without a major version increment. Changing file
-  formats also requires a major version number increment.
-- [ ] For substantial changes or changes to the command-line interface, is it
-  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
-  for more details.
```
Comment (standage) [inline .github/PULL_REQUEST_TEMPLATE.md]: These points are important, and have been included below.

**Comment by standage** [comment]
I would also argue that with this newer stripped-down checklist, it's reasonable to expect it to be completed before code review. As in "no Titus, I will not review your PR until you've ticked off the boxes."

**Comment by ctb** [comment]
will take a look - thanks.

--rod

**Comment by standage** [comment]
ping :-)

**Pull request merged**